File: ios/ExpoAdapterIterableModule.swift
import ExpoModulesCore
import IterableSDK
import UserNotifications

public class ExpoAdapterIterableModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoAdapterIterable")

    /**
     * Get the Iterable API key from the Info.plist file.
     * @return The Iterable API key.
     */
    Function("getApiKey") {
      return Bundle.main.object(forInfoDictionaryKey: "ITERABLE_API_KEY") as? String
    }

    AsyncFunction("requestNotificationPermission") { (promise: Promise) in
      UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .badge, .sound]) {
        (granted, error) in
        if let error = error {
          promise.reject(
            "ERR_NOTIFICATION",
            "Failed to request notification permission: \(error.localizedDescription)")
          return
        }
        promise.resolve(granted)
        ITBInfo("auth: \(granted)")
      }
    }
  }
}

File: android/src/main/java/expo/modules/adapters/iterable/ExpoAdapterIterableModule.kt
package expo.modules.adapters.iterable

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import android.content.pm.PackageManager

class ExpoAdapterIterableModule() : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoAdapterIterable")
  }
}